packages/api/models/collections.js

```diff
@@ -64,6 +65,41 @@ class Collection extends Manager {
 
     return super.create(item);
   }
+
+  /**
+   * Delete a collection
+   *
+   * @param {string} name - the collection name
+   * @param {string} version - the collection version
+   */
+  async delete(name, version) {
+    if (!(await this.exists(name, version))) throw new Error('Collection does not exist');
+
+    if (await this.hasAssociatedRules(name, version)) {
+      throw new Error('Cannot delete a collection that has associated rules');
+    }
+
+    await super.delete({ name, version });
+  }
+
+  /**
+   * Test if there are any rules associated with the collection
+   *
+   * @param {string} name - collection name
+   * @param {string} version - collection version
+   * @returns {Promise<boolean>}
+   */
+  async hasAssociatedRules(name, version) {
+    const ruleModel = new Rule();
+    const rules = (await ruleModel.scan()).Items;
```

Comment:
Is it not possible to use DynamoDB filter expressions rather than scanning the table and iterating over all the results?

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SQLtoNoSQL.ReadData.Query.html#SQLtoNoSQL.ReadData.Query.DynamoDB

Of course, I'm open to being told that you think this approach is preferable regardless if there's a reason for that

Reply:
Looking closer, I see that you are already using the `scan` method implemented on the base model, so is there some reason it wasn't possible to pass in filter conditions for the collection attributes?